resourcemanager/pool/spool/spool.go

```diff
@@ -197,3 +210,21 @@ func (p *Pool) ReleaseAndWait() {
 	p.wg.Wait()
 	resourcemanager.InstanceResourceManager.Unregister(p.Name())
 }
+
+func runTask(task *poolmanager.Meta) {
+	taskCh := task.GetTaskCh()
+	exitCh := task.GetExitCh()
+	task.IncTask()
+	defer task.DecTask()
+	for {
+		select {
+		case f, ok := <-taskCh:
+			if !ok {
+				return
+			}
+			f()
```

Comment:
Seems if `f()` runs a long time, the goroutine's shrink will be slow because it will waiting for `f()` exit. Is it better to pass a `Context` to f and then the goroutine is about to exit, the function can be notified.

Reply:
I agree with you. We can solve it in the next PR.